dockerfile/parser/parser.go

```diff
@@ -313,6 +352,39 @@ func Parse(rwc io.Reader) (*Result, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		if child.canContainHeredoc() {
+			heredocs, err := heredocsFromLine(line)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, heredoc := range heredocs {
+				terminator := []byte(heredoc.Name)
+				terminated := false
+				for scanner.Scan() {
+					bytesRead := scanner.Bytes()
+					currentLine++
+
+					possibleTerminator := trimNewline(bytesRead)
+					if heredoc.Chomp {
+						possibleTerminator = trimLeadingTabs(possibleTerminator)
+					}
+					if bytes.Equal(possibleTerminator, terminator) {
+						terminated = true
+						break
+					}
+					heredoc.Content += "\n"
+					heredoc.Content += string(bytesRead)
+				}
+				if !terminated {
+					return nil, errors.New("unterminated heredoc")
```

Comment:
I'd suggest including `heredoc.Name` in the error message.

Reply:
done